tests/goal/test_goal.py

```diff
@@ -80,6 +80,30 @@ def test_goal_simple_args() -> None:
     verify(result.output)
 
 
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_input_file() -> None:
+    result = invoke("goal clerk compile approval.teal")
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_output_file() -> None:
+    result = invoke("goal clerk compile approval.compiled")
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_input_output_files() -> None:
+    result = invoke("goal clerk compile approval.teal approval.compiled")
```

Comment:
when you don't provide -o flag it assumes all inputs after compile are batched inputs, make sure to test cases when you have several input files and a single output as well.